cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaAdminTopicConfigProvider.java

```diff
@@ -0,0 +1,171 @@
+ */
+public class KafkaAdminTopicConfigProvider extends JsonFileTopicConfigProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaAdminTopicConfigProvider.class);
+
+  private Properties _clusterConfigs;
+  private AdminClient _adminClient;
+
+  @Override
+  public Properties clusterConfigs() {
+    return _clusterConfigs;
+  }
+
+  /**
+   * Fetches the configuration for the requested topic. If an error is encountered the details will be logged and an
+   * empty Properties instance will be returned.
+   *
+   * @param topic Topic name for which the topic-level configurations are required.
+   * @return Properties instance containing the topic configuration.
+   */
+  @Override
+  public Properties topicConfigs(String topic) {
+    Config topicConfig = null;
+    ConfigResource topicResource = new ConfigResource(ConfigResource.Type.TOPIC, topic);
+    try {
+      LOG.debug("Requesting details for topic '{}'", topic);
+      topicConfig = _adminClient
+              .describeConfigs(Collections.singletonList(topicResource))
+              .all()
+              .get()
+              .get(topicResource);
+    } catch (ExecutionException ee) {
+      if (org.apache.kafka.common.errors.TimeoutException.class == ee.getCause().getClass()) {
+        LOG.warn("Failed to retrieve configuration for topic '{}' due to describeConfigs request time out. Check for Kafka-side issues"
+                + " and consider increasing the configured timeout.", topic);
+      } else {
+        // e.g. could be UnknownTopicOrPartitionException due to topic deletion or InvalidTopicException
+        LOG.warn("Cannot retrieve configuration for topic '{}'.", topic, ee);
+      }
+    } catch (InterruptedException ie) {
+      LOG.debug("Interrupted while getting configuration for topic '{}'.", topic, ie);
+    }
+
+    if (topicConfig != null) {
+      return convertTopicConfigToProperties(topicConfig);
+    } else {
+      LOG.warn("The configuration for topic '{}' could not be retrieved, returning empty Properties instance.", topic);
+      return new Properties();
+    }
+  }
+
+  /**
+   * Fetches the configuration for all the topics on the Kafka cluster. If an error is encountered when retrieving the
+   * topic names then the error details will be logged and an empty Map instance will be returned.
+   *
+   * @return A Map from topic name string to Properties instance containing that topic's configuration.
+   */
+  @Override
+  public Map<String, Properties> allTopicConfigs() {
+
+    // Request a map of futures for the config of each topic on the Kafka cluster
+    LOG.debug("Requesting configurations for all topics");
+    Set<String> topicNames = null;
+    try {
+      topicNames = _adminClient.listTopics().names().get();
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.warn("Unable to obtain list of all topic names from the Kafka Cluster");
+    }
+
+    if (topicNames == null) {
+      return new HashMap<>();
```

Comment:
Nit: `return new HashMap<>();` -> `return Collections.emptyMap();` -- i.e. no need to create an object.